pyquil/api/compiler.py

```diff
@@ -98,6 +97,9 @@ def compile(self, quil_program):
         :returns: The compiled Program object.
         :rtype: Program
         """
+        if self.specs is None and self.custom_isa is None:
+            raise ValueError("This connection doesn't have an associated ISA to target.")
```

Comment:
change to something more helpful. "You must provide at least either `device` or `isa_source` when constructing the compiler connection to use this method"